### PR TITLE
RAT-343: Add test to validate allowing a custom license

### PR DESCRIPTION
**Pull request by nielsbasjes**
A simple test to ensure custom licenses work

### CODE REVIEW

**Comment by nielsbasjes** [comment]
To make the test good I needed to remove the Apache license ... which triggered an NPE fixed in this separate merge request: https://github.com/apache/creadur-rat/pull/196

**Comment by nielsbasjes** [comment]
So there IS an NPE in the integration test but for some reason I do not know yet the build does not fail.

```
[INFO] --- invoker:3.6.0:integration-test (integration-test) @ apache-rat-plugin ---
[INFO] Building: RAT-268/pom.xml
[INFO]           RAT-268/pom.xml .................................. SUCCESS (3.393 s)
[INFO] Building: it1/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           it1/pom.xml ...................................... SUCCESS (3.249 s)
[INFO] Building: CustomLicense/pom.xml
[INFO]   The build exited with code 1. See /home/runner/work/creadur-rat/creadur-rat/apache-rat-plugin/target/invoker/CustomLicense/build.log for details.
[INFO]           CustomLicense/pom.xml ............................ FAILED (2.081 s)
[INFO] Building: it4_RAT-168/pom.xml
[INFO] run post-build script verify.groovy
[INFO]           it4_RAT-168/pom.xml .............................. SUCCESS (2.848 s)
```

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/Family.java [hunk 1]
```diff
@@ -38,4 +38,8 @@ public ILicenseFamily build() {
         return builder.setLicenseFamilyCategory(id).setLicenseFamilyName(name).build();
     }
 
+    @Override
+    public String toString() {
+        return '{' + id + ':' + name + '}';
+    }
```
Comment (Claudenw) [inline apache-rat-plugin/src/main/java/org/apache/rat/mp/Family.java:44]: Why do you need a toString() on the builder?  I don't see any uses of it in your code.

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/License.java [hunk 2]
```diff
@@ -57,4 +57,9 @@ public ILicense build(SortedSet<ILicenseFamily> context) {
                 .setName(name).setNotes(notes).build(context);
     }
 
+    @Override
+    public String toString() {
+        return '{' + family + ':' + id + ':' + name + '}';
+    }
+
```
Comment (Claudenw) [inline apache-rat-plugin/src/main/java/org/apache/rat/mp/License.java:64]: Why do you need a toString() on the builder?  I don't see any uses of it in your code.

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/Family.java (hunk 1 above)
Comment (nielsbasjes) [reply apache-rat-plugin/src/main/java/org/apache/rat/mp/Family.java:44]: Correct. I needed this to read the debug output when running mvn -X. Without this it shows the class name and the hash... which makes it useless output.

File: apache-rat-plugin/src/main/java/org/apache/rat/mp/License.java (hunk 2 above)
Comment (nielsbasjes) [reply apache-rat-plugin/src/main/java/org/apache/rat/mp/License.java:64]: Yes, same as the other one.

**Comment by ottlinger** [comment]
Thanks for the test and working example.

**Pull request merged**